File: Signal.CPP/IntroSpectral.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "IntroSpectral.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma resource "*.dfm"
TIntroSpectralForm *IntroSpectralForm;
//---------------------------------------------------------------------------
__fastcall TIntroSpectralForm::TIntroSpectralForm(TComponent* Owner)
	: TBasicForm3(Owner)
{
		RichEdit1->Lines->Clear();
		RichEdit1->DefAttributes->Size = 12;
		RichEdit1->DefAttributes->Style = RichEdit1->DefAttributes->Style << TFontStyle::fsBold;
		RichEdit1->SelAttributes->Color = (TColor) RGB(51,51,153);
		RichEdit1->SelAttributes->Size = 14;
		RichEdit1->Lines->Add("Frequency spectrum analysis features:");
		RichEdit1->Lines->Add("");
		RichEdit1->Paragraph->Numbering = nsBullet;
		RichEdit1->Paragraph->FirstIndent = 10;
		RichEdit1->Lines->Add("Very fast FFT for real and complex signals");
		RichEdit1->Lines->Add("Chirp-Z Transform for high speed frequency zoom-in");
		RichEdit1->Lines->Add("4 autoregressive methods for spectral estimation: "
				"Yulle-Walker, Burg, Covariance and Modified covariance");
		RichEdit1->Lines->Add("10 different time signal windows: Hanning, Hamming, FlatTop, "
				"Bartlet, Blackman, BlackmanHarris, BlackmanExact, "
				"Cosine-tappered, Kaiser, Chebyshev and  Exponential");
		RichEdit1->Lines->Add("THD, THD-N, SNR, NF, RMS, SFDR spectral statistics");
		RichEdit1->Lines->Add("Integration and differentiation in the frequency domain");
		RichEdit1->Lines->Add("Amplitude, power, peak-peak and RMS spectrums.");
		RichEdit1->Lines->Add("Real and complex cepstrum.");
		RichEdit1->Paragraph->Numbering = nsNone;
		RichEdit1->Paragraph->FirstIndent = 0;
}
//---------------------------------------------------------------------------

File: Signal.Delphi/IntroSpectral.pas
unit IntroSpectral;

interface

uses
  Windows, Messages, SysUtils, Classes, Graphics, Controls, Forms, Dialogs,
  Basic3, StdCtrls, ComCtrls;

type
  TIntroSpectralForm = class(TBasicForm3)
    procedure FormCreate(Sender: TObject);
  private
    { Private declarations }
  public
    { Public declarations }
  end;

var
  IntroSpectralForm: TIntroSpectralForm;

implementation

{$IFDEF CLR}
{$R *.nfm}
{$ELSE}
{$R *.dfm}
{$ENDIF}

procedure TIntroSpectralForm.FormCreate(Sender: TObject);
begin
  inherited;
  With RichEdit1.Lines, RichEdit1 do
  begin
    Clear;
    DefAttributes.Size := 10;
    DefAttributes.Style := [fsBold];
    SelAttributes.Size := 12;

    SelAttributes.Color := RGB(51,51,153);
    Add('Frequency spectrum analysis features:');
    Add('');
    Paragraph.FirstIndent := 10;
    Paragraph.Numbering := nsBullet;
    Add('Very fast FFT for real and complex signals');
    Add('Chirp-Z Transform for high speed frequency zoom-in');
    Add('4 autoregressive methods for spectral estimation: ' +
        'Yulle-Walker, Burg, Covariance and Modified covariance');
    Add('10 different time signal windows: Hanning, Hamming, FlatTop, ' +
        'Bartlet, Blackman, BlackmanHarris, BlackmanExact, '+
        'Cosine-tappered, Kaiser, Chebyshev and  Exponential');
    Add('THD, THD-N, SNR, NF, RMS, SFDR spectral statistics');
    Add('Integration and differentiation in the frequency domain');
    Add('Amplitude, power, peak-peak and RMS spectrums.');
    Add('Real and complex cepstrum.');

(*  Add('A comprehensive set of averaging methods: linear average, ' +
        'exponential average (= running average), peak hold average and ' +
        'averaging of the complex frequency spectrum.');
    Add('Compute RMS within frequency bands with bands editor.');
    Add('Computes standard deviation of the averaged frequency spectrum.');
    Add('Dedicated TSpectrumAnalyzer component with a component editor.');
    Add('Estimates the transfer function in amplitude and phase.'); *)
    Paragraph.Numbering := nsNone;
    Paragraph.FirstIndent := 0;
  end;
end;

initialization
   RegisterClass(TIntroSpectralForm);


end.
